### PR TITLE
feat: implement X data collection and processing pipeline for ranked influencers

### PR DESCRIPTION
**Pull request by B-Naoufal**
Add core tweet analysis functionality including:
- Tweet content validation with configurable length requirements
- Tweet processing pipeline with content cleaning
- Author name normalization from various formats
- Comprehensive test coverage for validation and processing

Details:
- Create Tweet and TweetAnalyzer structs for data handling
- Implement content cleaning to remove URLs, mentions, and special patterns
- Add validation logic for tweet content requirements
- Include table-driven tests covering edge cases and common scenarios
- Track valid/invalid tweets for analysis purposes
- Add stats collection functionality

Tests cover:
- Content validation with various lengths and formats
- Author name cleaning from different input formats
- Tweet processing with valid/invalid categorization
- Edge cases for content cleaning and validation

This commit sets up the foundation for tweet analysis pipeline, preparing for more advanced processing features.

### CODE REVIEW

File: finowl-backend/cmd/main.go [hunk 1]
```diff
@@ -1,11 +1,47 @@
...
+
+	// Start the bot
+	if err := bot.Start(); err != nil {
+		log.Fatal("Error starting bot:", err)
+	}
+
+	fmt.Println("Bot is now running. Press CTRL-C to exit.")
+
+	// Wait for interrupt signal
+	sc := make(chan os.Signal, 1)
```
Comment (finowl-near) [inline finowl-backend/cmd/main.go:41]: this just for testing purposes right ?

File: finowl-backend/pkg/analyzer/validator.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+import "strings"
+
+// ValidateTweetContent checks if the content meets our criteria
+func ValidateTweetContent(content string) bool {
+	// Clean the content first
+	cleanContent := CleanTweetContent(content)
+
+	// Check if content is empty after cleaning
+	if len(strings.TrimSpace(cleanContent)) == 0 {
```
Comment (finowl-near) [inline finowl-backend/pkg/analyzer/validator.go:11]: what's the goal of cleaning here ? fake tweets ?

**Review by finowl-near — CHANGES_REQUESTED** [review]
nice job for backend part , i tested it and fetched tweets correctly.

**Review by finowl-near — APPROVED** [review]

**Review by KD-ayoub — APPROVED** [review]
